kaupunginnaiset/alman-akka
# Add deployment to GitHub pages

## Purpose

* Add deployment to GitHub pages with custom domain almanakka.kaupunginnaiset.fi (triggers on each push to main)
* Needed to remove the use of next/image to enable static export. It there appears a need for image optimisations later, let's reconsider the deployment model (or write custom optimiser).
* Added a simple smoke test for the exported files via nginx-container. We could execute e2e-tests before deployment against this container once they are ready.
* Created a local action for setting up node according to @magdapoppins nvm idea, so that we don't have to duplicate the steps in each job.

## Related Issues

Closes #18 

## How to test

Check that CI-workflow succeeds. After merging, check that app is visible at https://almanakka.kaupunginnaiset.fi
